File: prisma/seed.ts
const { PrismaClient } = require('@prisma/client')
const bcrypt = require('bcryptjs')

const prisma = new PrismaClient()

async function main() {
  console.log('Seeding database...')

  // --- Seed Demo User ---
  const hashedPassword = await bcrypt.hash('password123', 10)
  const demoUser = await prisma.user.upsert({
    where: { email: 'demo@finpulse.app' },
    update: {},
    create: {
      id: 'user-demo',
      email: 'demo@finpulse.app',
      name: 'Demo User',
      passwordHash: hashedPassword,
    },
  })
  console.log('Seeded demo user:', demoUser.email)

  // --- Seed Categories ---
  // Parent categories
  const parentCategories = [
    { id: 'cat-income', name: 'Income', type: 'INCOME', icon: 'DollarSign', color: '#16a34a' },
    { id: 'cat-housing', name: 'Housing', type: 'EXPENSE', icon: 'Home', color: '#2563eb' },
    { id: 'cat-transport', name: 'Transport', type: 'EXPENSE', icon: 'Car', color: '#7c3aed' },
    { id: 'cat-food', name: 'Food', type: 'EXPENSE', icon: 'UtensilsCrossed', color: '#f59e0b' },
    { id: 'cat-shopping', name: 'Shopping', type: 'EXPENSE', icon: 'ShoppingBag', color: '#ec4899' },
    { id: 'cat-subscriptions', name: 'Subscriptions', type: 'EXPENSE', icon: 'CreditCard', color: '#8b5cf6' },
    { id: 'cat-health', name: 'Health', type: 'EXPENSE', icon: 'Heart', color: '#ef4444' },
    { id: 'cat-personal', name: 'Personal', type: 'EXPENSE', icon: 'User', color: '#14b8a6' },
    { id: 'cat-travel', name: 'Travel', type: 'EXPENSE', icon: 'Plane', color: '#0ea5e9' },
    { id: 'cat-financial', name: 'Financial', type: 'EXPENSE', icon: 'Landmark', color: '#6b7280' },
    { id: 'cat-savings-investing', name: 'Savings & Investing', type: 'EXPENSE', icon: 'PiggyBank', color: '#16a34a' },
  ]

  for (const cat of parentCategories) {
    await prisma.category.upsert({
      where: { id: cat.id },
      update: {},
      create: cat,
    })
  }
  console.log('Seeded parent categories')

  // Child categories
  const childCategories = [
    // Income subcategories
    { id: 'cat-income-salary', name: 'Salary', type: 'INCOME', icon: 'Briefcase', color: '#16a34a', parentId: 'cat-income' },
    { id: 'cat-income-bonus', name: 'Bonus', type: 'INCOME', icon: 'Gift', color: '#16a34a', parentId: 'cat-income' },
    { id: 'cat-income-interest', name: 'Interest', type: 'INCOME', icon: 'Percent', color: '#16a34a', parentId: 'cat-income' },
    { id: 'cat-income-investment-returns', name: 'Investment Returns', type: 'INCOME', icon: 'TrendingUp', color: '#16a34a', parentId: 'cat-income' },
    { id: 'cat-income-side', name: 'Side Income', type: 'INCOME', icon: 'Zap', color: '#16a34a', parentId: 'cat-income' },
    { id: 'cat-income-refunds', name: 'Refunds', type: 'INCOME', icon: 'RotateCcw', color: '#16a34a', parentId: 'cat-income' },

    // Housing subcategories
    { id: 'cat-housing-rent', name: 'Rent', type: 'EXPENSE', icon: 'Home', color: '#2563eb', parentId: 'cat-housing' },
    { id: 'cat-housing-utilities', name: 'Utilities', type: 'EXPENSE', icon: 'Zap', color: '#2563eb', parentId: 'cat-housing' },
    { id: 'cat-housing-internet', name: 'Internet', type: 'EXPENSE', icon: 'Wifi', color: '#2563eb', parentId: 'cat-housing' },
    { id: 'cat-housing-renters-insurance', name: 'Renters Insurance', type: 'EXPENSE', icon: 'Shield', color: '#2563eb', parentId: 'cat-housing' },

    // Transport subcategories
    { id: 'cat-transport-car-payment', name: 'Car Payment', type: 'EXPENSE', icon: 'Car', color: '#7c3aed', parentId: 'cat-transport' },
    { id: 'cat-transport-gas', name: 'Gas', type: 'EXPENSE', icon: 'Fuel', color: '#7c3aed', parentId: 'cat-transport' },
    { id: 'cat-transport-insurance', name: 'Insurance', type: 'EXPENSE', icon: 'Shield', color: '#7c3aed', parentId: 'cat-transport' },
    { id: 'cat-transport-parking', name: 'Parking', type: 'EXPENSE', icon: 'ParkingCircle', color: '#7c3aed', parentId: 'cat-transport' },
    { id: 'cat-transport-rideshare', name: 'Rideshare', type: 'EXPENSE', icon: 'Navigation', color: '#7c3aed', parentId: 'cat-transport' },

    // Food subcategories
    { id: 'cat-food-groceries', name: 'Groceries', type: 'EXPENSE', icon: 'ShoppingCart', color: '#f59e0b', parentId: 'cat-food' },
    { id: 'cat-food-restaurants', name: 'Restaurants', type: 'EXPENSE', icon: 'UtensilsCrossed', color: '#f59e0b', parentId: 'cat-food' },
    { id: 'cat-food-coffee', name: 'Coffee', type: 'EXPENSE', icon: 'Coffee', color: '#f59e0b', parentId: 'cat-food' },
    { id: 'cat-food-delivery', name: 'Food Delivery', type: 'EXPENSE', icon: 'Truck', color: '#f59e0b', parentId: 'cat-food' },

    // Shopping subcategories
    { id: 'cat-shopping-clothing', name: 'Clothing', type: 'EXPENSE', icon: 'Shirt', color: '#ec4899', parentId: 'cat-shopping' },
    { id: 'cat-shopping-electronics', name: 'Electronics', type: 'EXPENSE', icon: 'Monitor', color: '#ec4899', parentId: 'cat-shopping' },
    { id: 'cat-shopping-home', name: 'Home', type: 'EXPENSE', icon: 'Armchair', color: '#ec4899', parentId: 'cat-shopping' },
    { id: 'cat-shopping-amazon', name: 'Amazon', type: 'EXPENSE', icon: 'Package', color: '#ec4899', parentId: 'cat-shopping' },
    { id: 'cat-shopping-general', name: 'General', type: 'EXPENSE', icon: 'ShoppingBag', color: '#ec4899', parentId: 'cat-shopping' },

    // Subscriptions subcategories
    { id: 'cat-subscriptions-streaming', name: 'Streaming', type: 'EXPENSE', icon: 'Tv', color: '#8b5cf6', parentId: 'cat-subscriptions' },
    { id: 'cat-subscriptions-software', name: 'Software', type: 'EXPENSE', icon: 'Code', color: '#8b5cf6', parentId: 'cat-subscriptions' },
    { id: 'cat-subscriptions-gym', name: 'Gym', type: 'EXPENSE', icon: 'Dumbbell', color: '#8b5cf6', parentId: 'cat-subscriptions' },
    { id: 'cat-subscriptions-cloud-storage', name: 'Cloud Storage', type: 'EXPENSE', icon: 'Cloud', color: '#8b5cf6', parentId: 'cat-subscriptions' },

    // Health subcategories
    { id: 'cat-health-medical', name: 'Medical', type: 'EXPENSE', icon: 'Stethoscope', color: '#ef4444', parentId: 'cat-health' },
    { id: 'cat-health-dental', name: 'Dental', type: 'EXPENSE', icon: 'Smile', color: '#ef4444', parentId: 'cat-health' },
    { id: 'cat-health-pharmacy', name: 'Pharmacy', type: 'EXPENSE', icon: 'Pill', color: '#ef4444', parentId: 'cat-health' },
    { id: 'cat-health-vision', name: 'Vision', type: 'EXPENSE', icon: 'Eye', color: '#ef4444', parentId: 'cat-health' },

    // Personal subcategories
    { id: 'cat-personal-haircut', name: 'Haircut', type: 'EXPENSE', icon: 'Scissors', color: '#14b8a6', parentId: 'cat-personal' },
    { id: 'cat-personal-gifts', name: 'Gifts', type: 'EXPENSE', icon: 'Gift', color: '#14b8a6', parentId: 'cat-personal' },
    { id: 'cat-personal-education', name: 'Education', type: 'EXPENSE', icon: 'GraduationCap', color: '#14b8a6', parentId: 'cat-personal' },

    // Travel subcategories
    { id: 'cat-travel-flights', name: 'Flights', type: 'EXPENSE', icon: 'Plane', color: '#0ea5e9', parentId: 'cat-travel' },
    { id: 'cat-travel-hotels', name: 'Hotels', type: 'EXPENSE', icon: 'Building', color: '#0ea5e9', parentId: 'cat-travel' },
    { id: 'cat-travel-activities', name: 'Activities', type: 'EXPENSE', icon: 'Map', color: '#0ea5e9', parentId: 'cat-travel' },

    // Financial subcategories
    { id: 'cat-financial-cc-fees', name: 'Credit Card Fees', type: 'EXPENSE', icon: 'CreditCard', color: '#6b7280', parentId: 'cat-financial' },
    { id: 'cat-financial-bank-fees', name: 'Bank Fees', type: 'EXPENSE', icon: 'Landmark', color: '#6b7280', parentId: 'cat-financial' },
    { id: 'cat-financial-atm', name: 'ATM', type: 'EXPENSE', icon: 'Banknote', color: '#6b7280', parentId: 'cat-financial' },

    // Savings & Investing subcategories
    { id: 'cat-si-401k', name: '401k Contribution', type: 'EXPENSE', icon: 'PiggyBank', color: '#16a34a', parentId: 'cat-savings-investing' },
    { id: 'cat-si-hysa', name: 'HYSA Transfer', type: 'EXPENSE', icon: 'ArrowRightLeft', color: '#16a34a', parentId: 'cat-savings-investing' },
    { id: 'cat-si-brokerage', name: 'Brokerage Deposit', type: 'EXPENSE', icon: 'TrendingUp', color: '#16a34a', parentId: 'cat-savings-investing' },
  ]

  for (const cat of childCategories) {
    await prisma.category.upsert({
      where: { id: cat.id },
      update: {},
      create: cat,
    })
  }
  console.log('Seeded child categories')

  // --- Seed Accounts ---
  const accounts = [
    {
      id: 'acct-bofa-checking',
      userId: demoUser.id,
      name: 'BoFA Checking',
      type: 'CHECKING',
      institution: 'Bank of America',
      balance: 450000,
      color: '#2563eb',
    },
    {
      id: 'acct-sofi-hysa',
      userId: demoUser.id,
      name: 'SoFi HYSA',
      type: 'SAVINGS',
      institution: 'SoFi',
      balance: 1200000,
      interestRate: 4.5,
      color: '#16a34a',
    },
    {
      id: 'acct-amex-delta',
      userId: demoUser.id,
      name: 'Amex Delta Gold',
      type: 'CREDIT_CARD',
      institution: 'American Express',
      balance: -185000,
      creditLimit: 1000000,
      color: '#7c3aed',
    },
    {
      id: 'acct-robinhood',
      userId: demoUser.id,
      name: 'Robinhood',
      type: 'INVESTMENT',
      institution: 'Robinhood',
      balance: 2800000,
      color: '#f59e0b',
    },
    {
      id: 'acct-car-loan',
      userId: demoUser.id,
      name: 'Car Loan',
      type: 'LOAN',
      institution: 'Toyota Financial',
      balance: -1850000,
      interestRate: 4.9,
      color: '#ef4444',
    },
  ]

  for (const acct of accounts) {
    await prisma.account.upsert({
      where: { id: acct.id },
      update: {},
      create: acct,
    })
  }
  console.log('Seeded accounts')

  // --- Seed Goals ---
  const goals = [
    {
      id: 'goal-car-loan-payoff',
      userId: demoUser.id,
      name: 'Car Loan Payoff',
      type: 'DEBT_PAYOFF',
      targetAmount: 1850000,
      currentAmount: 0,
    },
    {
      id: 'goal-home-purchase',
      userId: demoUser.id,
      name: 'Home Purchase in India',
      type: 'SAVINGS',
      targetAmount: 5000000,
      currentAmount: 0,
    },
    {
      id: 'goal-macbook',
      userId: demoUser.id,
      name: 'MacBook Pro Fund',
      type: 'PURCHASE',
      targetAmount: 350000,
      currentAmount: 125000,
    },
    {
      id: 'goal-emergency-fund',
      userId: demoUser.id,
      name: 'Emergency Fund',
      type: 'EMERGENCY_FUND',
      targetAmount: 3000000,
      currentAmount: 800000,
    },
  ]

  for (const goal of goals) {
    await prisma.goal.upsert({
      where: { id: goal.id },
      update: {},
      create: goal,
    })
  }
  console.log('Seeded goals')

  // --- Seed Auto-Categorization Rules ---
  const rules = [
    // Food - Groceries
    { keyword: 'walmart', categoryId: 'cat-food-groceries', priority: 10 },
    { keyword: 'kroger', categoryId: 'cat-food-groceries', priority: 10 },
    { keyword: 'whole foods', categoryId: 'cat-food-groceries', priority: 10 },
    { keyword: 'trader joe', categoryId: 'cat-food-groceries', priority: 10 },
    { keyword: 'costco', categoryId: 'cat-food-groceries', priority: 10 },
    { keyword: 'aldi', categoryId: 'cat-food-groceries', priority: 10 },

    // Food - Restaurants
    { keyword: 'chipotle', categoryId: 'cat-food-restaurants', priority: 10 },
    { keyword: 'mcdonald', categoryId: 'cat-food-restaurants', priority: 10 },
    { keyword: 'chick-fil-a', categoryId: 'cat-food-restaurants', priority: 10 },

    // Food - Coffee
    { keyword: 'starbucks', categoryId: 'cat-food-coffee', priority: 10 },
    { keyword: 'dunkin', categoryId: 'cat-food-coffee', priority: 10 },

    // Food - Delivery
    { keyword: 'doordash', categoryId: 'cat-food-delivery', priority: 10 },
    { keyword: 'uber eats', categoryId: 'cat-food-delivery', priority: 10 },
    { keyword: 'grubhub', categoryId: 'cat-food-delivery', priority: 10 },

    // Transport
    { keyword: 'uber', categoryId: 'cat-transport-rideshare', priority: 5 },
    { keyword: 'lyft', categoryId: 'cat-transport-rideshare', priority: 10 },
    { keyword: 'shell oil', categoryId: 'cat-transport-gas', priority: 10 },
    { keyword: 'chevron', categoryId: 'cat-transport-gas', priority: 10 },
    { keyword: 'exxon', categoryId: 'cat-transport-gas', priority: 10 },

    // Shopping
    { keyword: 'amazon', categoryId: 'cat-shopping-amazon', priority: 5 },
    { keyword: 'target', categoryId: 'cat-shopping-general', priority: 5 },
    { keyword: 'apple.com', categoryId: 'cat-shopping-electronics', priority: 10 },
    { keyword: 'best buy', categoryId: 'cat-shopping-electronics', priority: 10 },

    // Subscriptions
    { keyword: 'netflix', categoryId: 'cat-subscriptions-streaming', priority: 10 },
    { keyword: 'spotify', categoryId: 'cat-subscriptions-streaming', priority: 10 },
    { keyword: 'hulu', categoryId: 'cat-subscriptions-streaming', priority: 10 },
    { keyword: 'youtube premium', categoryId: 'cat-subscriptions-streaming', priority: 10 },
    { keyword: 'icloud', categoryId: 'cat-subscriptions-cloud-storage', priority: 10 },
    { keyword: 'google storage', categoryId: 'cat-subscriptions-cloud-storage', priority: 10 },

    // Health
    { keyword: 'cvs', categoryId: 'cat-health-pharmacy', priority: 10 },
    { keyword: 'walgreens', categoryId: 'cat-health-pharmacy', priority: 10 },
  ]

  // Delete existing rules for demo user and recreate
  await prisma.categorizationRule.deleteMany({ where: { userId: demoUser.id } })
  for (const rule of rules) {
    await prisma.categorizationRule.create({
      data: {
        userId: demoUser.id,
        pattern: rule.keyword,
        categoryId: rule.categoryId,
        priority: rule.priority,
      },
    })
  }
  console.log('Seeded categorization rules')

  console.log('Database seeding complete!')
}

main()
  .catch((e) => {
    console.error('Seed error:', e)
    process.exit(1)
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

File: backend/seed.py
"""
Seed script — populates the database with demo data.

Usage (inside backend container):
    python seed.py

Demo credentials:
    Email:    demo@finpulse.app
    Password: password123
"""

from datetime import datetime, timezone, timedelta
from database import engine
from models import (
    SQLModel, User, Account, Category, Transaction,
    Budget, Goal, CategorizationRule,
)
from sqlmodel import Session, select
import bcrypt
import uuid


def uid():
    return str(uuid.uuid4())


def now():
    return datetime.now(timezone.utc)


def days_ago(n: int) -> datetime:
    return now() - timedelta(days=n)


def main():
    SQLModel.metadata.create_all(engine)

    with Session(engine) as session:
        # ── Demo user ────────────────────────────────────────────
        existing = session.exec(select(User).where(User.email == "demo@finpulse.app")).first()
        if existing:
            print("Demo user already exists — skipping seed.")
            return

        hashed = bcrypt.hashpw(b"password123", bcrypt.gensalt()).decode()
        user = User(
            id=uid(),
            email="demo@finpulse.app",
            name="Demo User",
            passwordHash=hashed,
            role="USER",
        )
        session.add(user)

        # ── Categories ───────────────────────────────────────────
        def cat(name, type_, icon=None, color="#6366f1", parent_id=None):
            c = Category(id=uid(), name=name, type=type_, icon=icon, color=color, parentId=parent_id)
            session.add(c)
            return c

        # Income
        income        = cat("Income",       "INCOME", "💰", "#22c55e")
        salary        = cat("Salary",        "INCOME", "💼", "#22c55e", income.id)
        freelance     = cat("Freelance",     "INCOME", "🧑‍💻", "#16a34a", income.id)
        investments   = cat("Investments",  "INCOME", "📈", "#15803d", income.id)

        # Expenses — housing
        housing       = cat("Housing",      "EXPENSE", "🏠", "#f59e0b")
        rent          = cat("Rent",         "EXPENSE", "🏠", "#f59e0b", housing.id)
        utilities     = cat("Utilities",    "EXPENSE", "💡", "#d97706", housing.id)

        # Expenses — food
        food          = cat("Food",         "EXPENSE", "🍔", "#ef4444")
        groceries     = cat("Groceries",    "EXPENSE", "🛒", "#ef4444", food.id)
        dining        = cat("Dining Out",   "EXPENSE", "🍽️",  "#dc2626", food.id)

        # Expenses — transport
        transport     = cat("Transport",    "EXPENSE", "🚗", "#8b5cf6")
        car_payment   = cat("Car Payment",  "EXPENSE", "🚙", "#7c3aed", transport.id)
        gas           = cat("Gas",          "EXPENSE", "⛽",  "#6d28d9", transport.id)

        # Expenses — subscriptions
        subscriptions = cat("Subscriptions","EXPENSE", "📱", "#06b6d4")
        streaming     = cat("Streaming",    "EXPENSE", "🎬", "#0891b2", subscriptions.id)

        # Expenses — health & personal
        health        = cat("Healthcare",   "EXPENSE", "🏥", "#f97316")
        personal      = cat("Personal",     "EXPENSE", "👤", "#ec4899")
        shopping      = cat("Shopping",     "EXPENSE", "🛍️",  "#a855f7")
        entertainment = cat("Entertainment","EXPENSE", "🎮", "#14b8a6")
        savings_cat   = cat("Savings",      "EXPENSE", "🏦", "#64748b")

        # ── Accounts ─────────────────────────────────────────────
        checking = Account(
            id=uid(), userId=user.id, name="BofA Checking",
            type="CHECKING", institution="Bank of America",
            balance=485320, color="#ef4444",
        )
        savings_acc = Account(
            id=uid(), userId=user.id, name="SoFi Savings",
            type="SAVINGS", institution="SoFi",
            balance=1250000, color="#22c55e",
        )
        credit = Account(
            id=uid(), userId=user.id, name="Amex Gold",
            type="CREDIT_CARD", institution="American Express",
            balance=-145600, creditLimit=500000, color="#f59e0b",
        )
        invest = Account(
            id=uid(), userId=user.id, name="Robinhood",
            type="INVESTMENT", institution="Robinhood",
            balance=3720000, color="#8b5cf6",
        )
        loan = Account(
            id=uid(), userId=user.id, name="Toyota Auto Loan",
            type="LOAN", institution="Toyota Financial",
            balance=-1540000, interestRate=4.9, color="#64748b",
        )
        for acc in [checking, savings_acc, credit, invest, loan]:
            session.add(acc)

        # ── Transactions ─────────────────────────────────────────
        def tx(account, amount, description, category, merchant=None, days=0):
            t = Transaction(
                id=uid(), userId=user.id, accountId=account.id,
                date=days_ago(days), amount=amount,
                description=description, categoryId=category.id,
                merchant=merchant,
            )
            session.add(t)

        # Income
        tx(checking,  895000, "Payroll Deposit",        salary,       "Acme Corp",      1)
        tx(checking,  895000, "Payroll Deposit",        salary,       "Acme Corp",     15)
        tx(checking,  895000, "Payroll Deposit",        salary,       "Acme Corp",     31)
        tx(checking,  150000, "Freelance Payment",      freelance,    "Client A",       5)
        tx(invest,     48200, "Dividend Income",        investments,  "Robinhood",     10)

        # Housing
        tx(checking, -180000, "Rent Payment",           rent,         "Landlord",       1)
        tx(checking,  -12400, "Electric Bill",          utilities,    "PG&E",           3)
        tx(checking,   -8900, "Internet",               utilities,    "Comcast",        5)

        # Food
        tx(credit,    -23400, "Whole Foods Market",     groceries,    "Whole Foods",    2)
        tx(credit,    -15600, "Trader Joe's",           groceries,    "Trader Joes",    6)
        tx(credit,    -8900,  "Chipotle",               dining,       "Chipotle",       3)
        tx(credit,    -14200, "Nobu Restaurant",        dining,       "Nobu",           8)
        tx(credit,    -6700,  "Starbucks",              dining,       "Starbucks",      1)
        tx(credit,    -5400,  "Uber Eats",              dining,       "Uber Eats",      4)

        # Transport
        tx(checking,  -45000, "Car Payment",            car_payment,  "Toyota",         1)
        tx(credit,    -7200,  "Shell Gas Station",      gas,          "Shell",          4)
        tx(credit,    -6800,  "Chevron",                gas,          "Chevron",       12)

        # Subscriptions
        tx(credit,    -1599,  "Netflix",                streaming,    "Netflix",        5)
        tx(credit,    -1399,  "Spotify",                streaming,    "Spotify",        5)
        tx(credit,    -1499,  "Disney+",                streaming,    "Disney Plus",    5)
        tx(credit,   -14999,  "GitHub Copilot",         subscriptions,"GitHub",         2)

        # Health
        tx(checking,  -25000, "Doctor Visit Copay",     health,       "Kaiser",         7)
        tx(credit,    -4500,  "CVS Pharmacy",           health,       "CVS",            9)

        # Shopping
        tx(credit,    -34900, "Amazon Order",           shopping,     "Amazon",         6)
        tx(credit,    -18900, "Target",                 shopping,     "Target",        11)

        # Entertainment
        tx(credit,    -7500,  "AMC Movie Tickets",      entertainment,"AMC",           13)
        tx(credit,    -12000, "Concert Tickets",        entertainment,"Ticketmaster",  18)

        # Savings
        tx(savings_acc,-100000,"Transfer to Savings",   savings_cat,  None,             1)
        tx(savings_acc,-100000,"Transfer to Savings",   savings_cat,  None,            31)

        # ── Budgets ──────────────────────────────────────────────
        budgets = [
            (groceries,     60000),
            (dining,        40000),
            (entertainment, 20000),
            (shopping,      50000),
            (utilities,     25000),
            (health,        30000),
        ]
        for category, amount in budgets:
            session.add(Budget(
                id=uid(), userId=user.id, categoryId=category.id,
                amount=amount, period="MONTHLY",
            ))

        # ── Goals ────────────────────────────────────────────────
        goals = [
            ("Emergency Fund",     1000000,  800000,  "EMERGENCY_FUND", savings_acc.id, 180),
            ("Down Payment",       5000000, 1250000,  "HOME_PURCHASE",  savings_acc.id, 730),
            ("Vacation Fund",       300000,   85000,  "VACATION",       None,            90),
            ("Investment Target",  5000000, 3720000,  "INVESTMENT",     invest.id,      365),
        ]
        for name, target, current, type_, linked_id, days in goals:
            session.add(Goal(
                id=uid(), userId=user.id, name=name,
                targetAmount=target, currentAmount=current,
                type=type_, linkedAccountId=linked_id,
                deadline=now() + timedelta(days=days),
            ))

        # ── Categorization rules ─────────────────────────────────
        rules = [
            ("Whole Foods",   groceries.id,   False),
            ("Trader Joe",    groceries.id,   False),
            ("WHOLEFDS",      groceries.id,   False),
            ("Chipotle",      dining.id,      False),
            ("Starbucks",     dining.id,      False),
            ("Uber Eats",     dining.id,      False),
            ("Netflix",       streaming.id,   False),
            ("Spotify",       streaming.id,   False),
            ("Disney",        streaming.id,   False),
            ("Amazon",        shopping.id,    False),
            ("Target",        shopping.id,    False),
            ("Shell",         gas.id,         False),
            ("Chevron",       gas.id,         False),
            ("CVS",           health.id,      False),
            ("Payroll",       salary.id,      False),
            (r"Payroll\s+Deposit", salary.id, True),
        ]
        for pattern, cat_id, is_regex in rules:
            session.add(CategorizationRule(
                id=uid(), userId=user.id, pattern=pattern,
                categoryId=cat_id, isRegex=is_regex,
            ))

        session.commit()
        print("✓ Demo data seeded successfully.")
        print("  Login: demo@finpulse.app / password123")


if __name__ == "__main__":
    main()
